tests/shell/shell_recorder/globaltest.dat

```diff
@@ -0,0 +1,34 @@
+Storage: dump
+Mountpoint: system/testmount/list
+File: /tmp/listtest.dump
+MountArgs: 
+DiffType: File
+
+
+< set system/elektra/globalplugins
+< set system/elektra/globalplugins/postcommit list
+< set system/elektra/globalplugins/postcommit/user
+< set system/elektra/globalplugins/postcommit/user/placements
+< set system/elektra/globalplugins/postcommit/user/placements/set "presetstorage precommit postcommit"
+< set system/elektra/globalplugins/postcommit/user/placements/get "pregetstorage postgetstorage"
+< set system/elektra/globalplugins/postcommit/user/placements/error "prerollback postrollback"
+< set system/elektra/globalplugins/postcommit/user/plugins
+< set system/elektra/globalplugins/postcommit/user/plugins/#0 tracer 
+< set system/elektra/globalplugins/postcommit/user/plugins/#0/placements
+< set system/elektra/globalplugins/postcommit/user/plugins/#0/placements/set "presetstorage precommit postcommit"
+< set system/elektra/globalplugins/postcommit/user/plugins/#1 timeofday 
+< set system/elektra/globalplugins/postcommit/user/plugins/#1/placements
+< set system/elektra/globalplugins/postcommit/user/plugins/#1/placements/set "presetstorage precommit postcommit"
+
+
+< set system/elektra/globalplugins/postrollback list
+< set system/elektra/globalplugins/precommit list
+< set system/elektra/globalplugins/pregetstorage list
+< set system/elektra/globalplugins/postgetstorage list 
+< set system/elektra/globalplugins/presetstorage list
+< set system/elektra/globalplugins/prerollback list
+
+STDOUT: (tracer.*){3}
+STDERR: (set.*di.*pos.*){3}
+RET: 0
+< set system/globaltestkey
```

Comment:
STDOUT/STDERR/RET/WARNINGS/ERRORS/DIFF always refer to the next command (< ...)
Their syntax is posix extended regex and they are matched against the output/return values of the next command.

The only thing that needs to be specified is "Mountpoint" because it's used to lookup the configuration file, for exports/comparisons, and for backups/restoring of the original state.

i've added some more examples:

the shell recorder creates a copy of the configuration/key file before running a command and then diffs them to keep track of all changes. 
db_changes.dat is an example of the "DIFF"-Operation which can be used to check if the changes are what we expected.


replay_test_fail.dat and replay_test_succeeds.dat are both examples of the replay function.
if the script contains the string "REPLAY!" the shell recorder runs the script twice, keeps a copy of the protocol and the configuration file from the first iteration, and compares them against the results from the second one.